### DISCUSSION TITLE
Starting from version 7.44.0, required attribute is no longer being set on inputs

### DISCUSSION THREAD

**Discussion by louh** [react-hook-form/react-hook-form#10655] [Q&A] [answered]
I have a component that has a required input field, like this:

```
<input
  type="email"
  name="email"
  placeholder="test@example.com"
  {...register('email', { required: true })}
/>
```

The expected output should be:
```
<input
  name="email"
  placeholder="test@example.com"
  required=""
  type="email"
/>
```

However, starting from version 7.44.0 onwards, the required attribute is being dropped from the rendered output:

```
<input
  name="email"
  placeholder="test@example.com"
  type="email"
/>
```

The component does use native validation (I already looked at https://github.com/orgs/react-hook-form/discussions/10596)

```
  const { register, handleSubmit } = useForm({
    shouldUseNativeValidation: true
  })
```

Is there another reason why `required` is longer being applied? Is something failing with the native validation option?

**Comment by leapful** [comment, marked as answer]
@louh 

You need to use option `progressive=true` to pass down native validation rule attributes.

Reference: https://react-hook-form.com/docs/useform/form#rules


Working demo: https://codesandbox.io/s/determined-northcutt-6vwsss?file=/src/App.tsx

**Comment by louh** [comment]
That did solve it -- I'm not exactly sure why this change is not a breaking change, however.
